stuffer/s2n_stuffer.c

```diff
@@ -50,11 +50,13 @@ int s2n_stuffer_init(struct s2n_stuffer *stuffer, struct s2n_blob *in)
 }
 int s2n_stuffer_alloc(struct s2n_stuffer *stuffer, const uint32_t size)
 {
+    notnull_check(stuffer);
```

Comment:
Would it make sense to zero out the stuffer on failure?

Reply:
@camshaft 

Would it make sense to do something like this:

```
    notnull_check(stuffer);
    *stuffer = (struct s2n_stuffer) {0};
    GUARD(s2n_alloc(&stuffer->blob, size));
    GUARD(s2n_stuffer_init(stuffer, &stuffer->blob));

    stuffer->alloced = 1;

    POSTCONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
    return S2N_SUCCESS;
```

Reply:
@camshaft any suggestions?

Reply:
That's definitely safer in the case that `s2n_stuffer_init` doesn't actually set all of the fields. I like it!

Reply:
Great! Implemented as suggested :)